src/cpp/src/whisper_pipeline_static.cpp

```diff
@@ -340,16 +542,57 @@ WhisperPipeline::StaticWhisperPipeline::StaticWhisperPipeline(const std::filesys
     auto decoder_model = core.read_model(models_path / "openvino_decoder_model.xml");
     auto decoder_with_past_model = core.read_model(models_path / "openvino_decoder_with_past_model.xml");
 
+    add_attention_mask_input_for_decoder(decoder_model);
+    add_attention_mask_input(decoder_with_past_model);
+
     // TODO: Support models produced by optimum-cli
     if (!check_decoder_model_compatibility(decoder_model)) {
         OPENVINO_THROW("StaticWhisperPipeline expects decoder model has \"attention_mask\" input!");
     }
 
     // TODO: There must be model reshape to eliminate dynamism!
-
-    m_models.encoder = core.compile_model(encoder_model, "NPU").create_infer_request();
-    m_models.decoder = core.compile_model(decoder_model, "NPU").create_infer_request();
-    m_models.decoder_with_past = core.compile_model(decoder_with_past_model, "NPU").create_infer_request();
+    size_t max_sequence_length = 128;
+
+    reshape_to_static_encoder(encoder_model);
+    reshape_to_static(decoder_model, 4, 4);  // What is 4 here??
```

Comment:
4 is max number of input tokens for decoder

Reply:
This comment should be removed

Reply:
Done